openerp/fields.py

```diff
@@ -253,56 +257,122 @@ class Second(models.Model):
+        'compute',              # compute(recs) computes field on recs
+        'compute_sudo',         # whether field should be recomputed as admin
+        'inverse',              # inverse(recs) inverses field on recs
+        'search',               # search(recs, operator, value) searches on self
+        'related',              # sequence of field names, for related fields
+        'related_sudo',         # whether related fields should be read as admin
+        'company_dependent',    # whether `self` is company-dependent (property field)
+        'default',              # default(recs) returns the default value
+
+        'string',               # field label
+        'help',                 # field tooltip
+        'readonly',             # whether the field is readonly
+        'required',             # whether the field is required
+        'states',               # set readonly and required depending on state
+        'groups',               # csv list of group xml ids
+        'change_default',       # whether the field may trigger a "user-onchange"
+        'deprecated',           # whether the field is deprecated
+
+        'inverse_fields',       # collection of inverse fields (objects)
+        'computed_fields',      # fields computed with the same method as self
+        'related_field',        # corresponding related field
+        '_triggers',            # invalidation and recomputation triggers
+    ]
+
+    # default values for slots
+    _default_automatic = False
+    _default_inherited = False
+    _default_column = None
+    _default_setup_done = False
+
+    _default_name = None
+    _default_model_name = None
+    _default_comodel_name = None
+
+    _default_store = True
+    _default_index = False
+    _default_manual = False
+    _default_copy = True
+    _default_depends = ()
+    _default_recursive = False
+    _default_compute = None
+    _default_compute_sudo = False
+    _default_inverse = None
+    _default_search = None
+    _default_related = None
+    _default_related_sudo = True
+    _default_company_dependent = False
+    _default_default = None
+
+    _default_string = None
+    _default_help = None
+    _default_readonly = False
+    _default_required = False
+    _default_states = None
+    _default_groups = False
+    _default_change_default = None
+    _default_deprecated = None
+
+    _default_inverse_fields = ()
+    _default_computed_fields = ()
+    _default_related_field = None
+    _default__triggers = ()
 
     def __init__(self, string=None, **kwargs):
         kwargs['string'] = string
-        self._attrs = {key: val for key, val in kwargs.iteritems() if val is not None}
-        self._free_attrs = []
+        attrs = {key: val for key, val in kwargs.iteritems() if val is not None}
+        self._attrs = attrs or EMPTY_DICT
+
+    def __getattr__(self, name):
+        """ Access non-slot field attribute. """
+        if name == '_attrs':
+            raise AttributeError(name)
```

Comment:
I think that's never going to be triggered (and it would prevent 356 if it did), `__getattr__` is triggered at the very end of the lookup if nothing has matched. `_attrs` is a true attribute.